torch/_functorch/aot_autograd.py

```diff
@@ -602,24 +602,23 @@ def convert(idx, x):
                     if req_subclass_dispatch:
                         fw_metadata = run_functionalized_fw_and_collect_metadata(
                             flat_fn,
-                            keep_input_mutations=aot_config.keep_inference_input_mutations
-                            and not needs_autograd,
-                            is_train=needs_autograd,
+                            keep_input_mutations=aot_config.keep_inference_input_mutations,
+                            is_train=False,
                             pre_dispatch=aot_config.pre_dispatch,
                         )(*fake_flat_args)
                     else:
                         fw_metadata = ViewAndMutationMeta(
                             input_info=fw_metadata.input_info,
                             output_info=fw_metadata.output_info,
                             num_intermediate_bases=fw_metadata.num_intermediate_bases,
-                            keep_input_mutations=aot_config.keep_inference_input_mutations
-                            and not needs_autograd,
+                            keep_input_mutations=aot_config.keep_inference_input_mutations,
                             traced_tangents=fw_metadata.traced_tangents,
                             subclass_inp_meta=fw_metadata.subclass_inp_meta,
                             subclass_fw_graph_out_meta=fw_metadata.subclass_fw_graph_out_meta,
                             subclass_tangent_meta=fw_metadata.subclass_tangent_meta,
-                            is_train=needs_autograd,
+                            is_train=False,
                             tokens=fw_metadata.tokens,
+                            static_parameter_indices=fw_metadata.static_parameter_indices,
```

Comment:
this adds the static params into the reconstructed metadata. above lines are just logic simplifications since line 596 sets needs_autograd to False